day_09.md

```diff
@@ -0,0 +1,265 @@
+
+That's something you are used to do with other people, isn't it?
+
+#### Syntax
+
+```javascript
+  new Promise(executor);
+  // or
+  new Promise((resolve, reject) => {
+    executor body
+  });
+```
+
+Well, Promises work kinda that.
+
+```javascript
+  new Promise((resolve, reject) => {
+      console.log('I started');
+      resolve();
+  })
+  .then(() => {
+      console.log(`A`);
+      throw new Error('I failed in A');
+  })
+  .then(() => {
+      console.log(`B`);
+  })
+  .catch((reason) => {
+      console.log(reason);
+  })
+  .finally(() => {
+      console.log(`C`);
+      console.log(`D`);
+  });
+
+```
+
+Easy to read, easy to control ... sort of, but definitely better than callback hell.
+
+Too much talking, let go to the best sources you can get.
+
+- [ECMAScript Promise Objects](https://www.ecma-international.org/ecma-262/6.0/#sec-promise-objects)
+- [ECMAScript Promise Constructor](https://www.ecma-international.org/ecma-262/6.0/#sec-promise-constructor)
+- [MDN - Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+- [MDN - Using Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises)
+- [YDKJS - Async & Performance - Chapter 3: Promises](https://github.com/getify/You-Dont-Know-JS/blob/1st-ed/async%20%26%20performance/ch3.md) | by Kyle Simpson
+
+---
+
+### Async/await
+
+Wait what? we were talking about ES2015 but this is not an ES2015 feature, nor ES2016!! We had to wait until ES2017 (ES8?? nope, remember that's not the way to name ES versions anymore) to have it.
+Then why are we talking about this? Simply because is one of the most important tools (together with generators and promises) to write asynchronous programs in JavaScript today.
+
+Imagine there's a feature to handle under the hood a the event loop and the promises to **operate asynchronously BUT resembles synchronous** code on its syntax. Since many program errors are originated on the difficulty of understanding a complex structure (e.g. like a promise chain) this feature might help with that, isn't it?
+
+First things first. If we go to the [ECMA2017 - 9.2 - ECMAScript Function Objects definition](https://www.ecma-international.org/ecma-262/8.0/index.html#sec-ecmascript-function-objects) we'll find that `async` is one of the 4 kinds of functions `normal`, `classConstructor`, `generator`and `async`, and if you search "async" there list of results is around 450. There's a lot to read there if you want; I'd suggest you start reading ad the [14.6 - Async Function Definition](https://www.ecma-international.org/ecma-262/8.0/index.html#sec-async-function-definitions).
```

Comment:
missed white-space: "is one of the 4 kinds of functions `normal`, `classConstructor`, **`generator`and** `async`".
Note: I noticed it is showing as highlighted and it has extra margin however it is showing together here. 

Could it be use: "and if you search **for** "async" **the** list of results is around 450" instead of "and if you search "async" **there** list of results is around 450"

Could it be use: "There's a lot to read if you want **to**; I'd suggest you **to** start reading **at** the" instead of "There's a lot to read there if you want; I'd suggest you start reading ad the"